analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ClearTextProtocolsAreSensitive.cs

```diff
@@ -61,16 +55,54 @@ public sealed class ClearTextProtocolsAreSensitive : HotspotDiagnosticAnalyzer
             {"clear-text SMTP", "SMTP over SSL/TLS or SMTP with STARTTLS" }
         };
 
+        private readonly List<string> commonlyUsedXmlDomains = new List<string>
+        {
+            "www.w3.org",
+            "xml.apache.org",
+            "schemas.xmlsoap.org",
+            "schemas.openxmlformats.org",
+            "rdfs.org",
+            "purl.org",
+            "xmlns.com",
+            "schemas.google.com",
+            "a9.com",
+            "ns.adobe.com",
+            "ltsc.ieee.org",
+            "docbook.org",
+            "graphml.graphdrawing.org",
+            "json-schema.org"
+        };
+
+        private readonly List<string> commonlyUsedExampleDomains = new List<string> {"example.com", "example.org", "test.com"};
+        private readonly List<string> localhostAddresses = new List<string> {"localhost", "127.0.0.1", "::1"};
+
         private readonly CSharpObjectInitializationTracker objectInitializationTracker =
             new CSharpObjectInitializationTracker(constantValue => constantValue is bool value && value,
                                                   ImmutableArray.Create(KnownType.System_Net_Mail_SmtpClient, KnownType.System_Net_FtpWebRequest),
                                                   propertyName => propertyName == EnableSslName);
 
+        private readonly Regex httpRegex;
+        private readonly Regex ftpRegex;
+        private readonly Regex telnetRegex;
+        private readonly Regex telnetRegexForIdentifier;
+        private readonly Regex validServerRegex;
+
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, EnableSslRule);
 
         public ClearTextProtocolsAreSensitive() : this(AnalyzerConfiguration.Hotspot) { }
 
-        public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration) { }
+        public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration)
+        {
+            const string allSubdomainsPattern = "([^/]+\\.)?";
```

Comment:
What's the idea behind `\` in `([^/]+\.)?`? To avoid overrunning to path part of uri in cases like `http://badidea.com/example.com` or `http://badidea.com/path?domain=example.com`? 
Let's add unit tests for such scenarios.

Reply:
If yes, what about `http://badidea.com?domain=example.com`? We should avoid `/?#@:`. Let's add UTs for all of them

Reply:
I do not see why @ and : are bad. Basically, this way you can specify the login and password inside the url.

Seems to be valid "example" url `http://user:passwd@example.com`. Wdyt?

Reply:
You're right, `:` and `@` are fine